src/main/scala/scalapb/json4s/JsonFormat.scala

```diff
@@ -387,7 +398,7 @@ class Printer private (config: Printer.PrinterConfig) {
           case Some(writer) => writer(this, e)
```

Comment:
if you provide an enum formatter such that it matches on Some you get performance degradation